sklearn/ensemble/iforest.py

```diff
@@ -181,10 +186,37 @@ def fit(self, X, y=None, sample_weight=None):
         super(IsolationForest, self)._fit(X, y, max_samples,
                                           max_depth=max_depth,
                                           sample_weight=sample_weight)
+
+        self.threshold_ = -sp.stats.scoreatpercentile(
+            -self.decision_function(X), 100. * (1. - self.contamination))
+
         return self
 
     def predict(self, X):
-        """Predict anomaly score of X with the IsolationForest algorithm.
+        """Outlyingness of observations in X according to the fitted model.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix of shape (n_samples, n_features)
+            The input samples. Internally, it will be converted to
+            ``dtype=np.float32`` and if a sparse matrix is provided
+            to a sparse ``csr_matrix``.
+
+        Returns
+        -------
+        is_inlier : array, shape = (n_samples, )
+            For each observations, tells whether or not (+1 or -1) it should
+            be considered as an outlier according to the fitted model.
+
+        """
+        X = check_array(X, accept_sparse='csr')
+        is_inlier = np.ones(X.shape[0], dtype=int)
+        is_inlier[self.decision_function(X) <= self.threshold_] = -1
```

Comment:
I was thinking more about the memory efficiency. @MechCoder your thoughts on this?